hazelcast/projection.py

```diff
@@ -0,0 +1,95 @@
+from hazelcast.serialization.api import IdentifiedDataSerializable
+
+_PROJECTIONS_FACTORY_ID = -30
+
+class Projection(object):
+    """Marker base class for all projections.
+
+    Projections allow the client to transform each query result object in order
+    to avoid redundant network traffic. The computation is performed in a fully
+    distributed manner, so no data other than the computed value is transferred
+    to the client, making the computation fast.
+    """
+    pass
+
+
+class _AbstractProjection(Projection, IdentifiedDataSerializable):
+    def write_data(self, object_data_output):
+        raise NotImplementedError("write_data")
+
+    def read_data(self, object_data_input):
+        pass
+
+    def get_factory_id(self):
+        return _PROJECTIONS_FACTORY_ID
+
+    def get_class_id(self):
+        raise NotImplementedError("get_class_id")
+
+
+class _SingleAttributeProjection(_AbstractProjection):
+    def __init__(self, attribute_path):
+        self._attribute_path = attribute_path
```

Comment:
Let's add the checks we have in Java client. Please define a utility function just above this class as we will reuse it.

```py
def _validate_attribute_path(attribute_path):
        if not attribute_path:
            raise ValueError("attribute_path must not be None or empty")
        
        if "[any]" in attribute_path:
            raise ValueError("attribute_path must not contain [any] operators")
```


```suggestion
        _validate_attribute_path(attribute_path)
        self._attribute_path = attribute_path
```